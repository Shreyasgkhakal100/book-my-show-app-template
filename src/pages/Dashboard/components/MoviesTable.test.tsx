import { render, renderWithRouter } from '@test-utils';
import { MoviesTable } from './';

const movies = [
    {
        id: 1,
        title: 'Movie 1',
        duration: 120
    },
    {
        id: 2,
        title: 'Movie 2',
        duration: 120
    },
    {
        id: 3,
        title: 'Movie 3',
        duration: 120
    }
];

describe('MoviesTable', () => {
    it('should render properly', () => {
        const { container } = renderWithRouter(<MoviesTable movies={movies} />);
        expect(container).toMatchSnapshot();
    });

    it('should render only thead & empty tbody tag for empty movies list', () => {
        const { container } = render(<MoviesTable movies={[]} />);
        expect(container).toMatchSnapshot();
    });
});
