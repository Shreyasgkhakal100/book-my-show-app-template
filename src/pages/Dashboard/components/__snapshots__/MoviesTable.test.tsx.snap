// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesTable should render only thead & empty tbody tag for empty movies list 1`] = `
<div>
  <table
    class="table"
  >
    <thead
      class="thead-light"
    >
      <tr>
        <th
          scope="col"
        >
          Sr. No.
        </th>
        <th
          scope="col"
        >
          Id
        </th>
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          Duration
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`MoviesTable should render properly 1`] = `
<div>
  <table
    class="table"
  >
    <thead
      class="thead-light"
    >
      <tr>
        <th
          scope="col"
        >
          Sr. No.
        </th>
        <th
          scope="col"
        >
          Id
        </th>
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          Duration
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          1
        </th>
        <td>
          1
        </td>
        <td>
          Movie 1
        </td>
        <td>
          120
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          2
        </th>
        <td>
          2
        </td>
        <td>
          Movie 2
        </td>
        <td>
          120
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          3
        </th>
        <td>
          3
        </td>
        <td>
          Movie 3
        </td>
        <td>
          120
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
